OracleHTTPServer/dockerfiles/12.2.1.2.0/Dockerfile

```diff
@@ -4,84 +4,97 @@
-    PATH=$PATH:/usr/java/default/bin:/u01/oracle/ohssa/oracle_common/common/bin
+    PATH=$PATH:/usr/java/default/bin:/u01/oracle/ohssa/oracle_common/common/bin \
+    NM_PORT=5556 \
+    OHS_LISTEN_PORT=7777 \
+    OHS_SSL_PORT=4443 \
+    JAVA_HOME=/usr/java/default \
+    MW_HOME=/u01/oracle/ohssa \
+    DOMAIN_NAME=ohsDomain \
+    OHS_COMPONENT_NAME=ohs_sa1 \
+    PATH=$PATH:$ORACLE_HOME/oracle_common/common/bin:$ORACLE_HOME/user_projects/domains/ohsDomain/bin:/u01/oracle/container-scripts
+
 
 # Copy packages to the new drive
 # -------------------------------------------------------
 COPY $OHS_PKG install.file oraInst.loc /u01/
 
-# Install OL7 required packages. Refer FMW 12.2.1 System requirements guide for complete list of packages
+# Install OL7 required packages. Refer FMW 12.2.1.2 System requirements guide for complete list of packages
 # Change the open file limits and kernel parameters that need changing
 # Setup filesystem and oracle user
 #---------------------------------------------------------------------------------------------------------
 # Proxy needs to be set before running yum in /etc/yum.conf
 RUN yum install -y \
-        unzip tar nfs-utils binutils compat-libcap1 compat-libstdc++ \
-        gcc gcc-c++ glibc glibc-devel libaio libaio-devel libgcc \
-	libstdc++ libstdc++-devel ksh make ocfs2-tools sysstat && \
+        unzip libaio sysstat && \
     yum clean all && \
-    rm -rf /var/cache/yum/* && \
     chmod a+xr /u01 && \
     useradd -b /u01 -m -s /bin/bash oracle && \
-    echo oracle:oracle | chpasswd && \
-    mkdir /u01/oracle/.inventory
+    mkdir /u01/oracle/.inventory && \
+    chown oracle:oracle -R /u01 
 
+# Install and configure Oracle JDK
 # Go to /u01 as user 'oracle' to proceed with OHS installation
 #--------------------------------------------------------
-RUN unzip -q -d /u01 /u01/$OHS_PKG && \
-    chown oracle:oracle -R /u01 && \
-    su -c "/u01/$OHS_BIN -ignoreSysPrereqs -novalidation -silent -responseFile /u01/install.file -invPtrLoc /u01/oraInst.loc -jreLoc $JAVA_HOME ORACLE_HOME=$ORACLE_HOME" - oracle && \
-    rm /u01/$OHS_PKG /u01/$OHS_BIN /u01/fmw_*_readme.htm /u01/oraInst.loc /u01/install.file
-
 USER oracle
-WORKDIR /u01/oracle/
+RUN cd /u01 && unzip -q /u01/$OHS_PKG && cd - && \
+    /u01/$OHS_BIN -ignoreSysPrereqs -novalidation -silent -responseFile /u01/install.file -invPtrLoc /u01/oraInst.loc ORACLE_HOME=$ORACLE_HOME && \ 
+    rm /u01/$OHS_BIN /u01/$OHS_PKG /u01/oraInst.loc /u01/install.file && \
+    mkdir /u01/oracle/container-scripts/ /u01/oracle/logs 
+
+# Copy required files to build this image
+# ------------------------------------------------------
+COPY container-scripts/* /u01/oracle/container-scripts/
+
+#Expose all Ports
+EXPOSE $OHS_SSL_PORT $OHS_LISTEN_PORT $NM_PORT
+
+# Final setup
+WORKDIR /u01/oracle/container-scripts
+
+# Create Doamin, OHS instance and start the  OHS instance
+RUN sh createDomain.sh 
```

Comment:
Do you need to do this during build? Can this be shifted into the `CMD` script instead so that the domain is created at run time? This would also allow you to have different configuration of the domain across individual containers without having to rebuild the image.